packages/contracts/src/box-objects.ts

```diff
@@ -0,0 +1,117 @@
+import { ErgoTree, OutputBuilder } from "@fleet-sdk/core";
+import { SByteType, SCollType, SConstant, SIntType, STupleType } from "@fleet-sdk/serializer";
+import { compile } from "@fleet-sdk/compiler";
+import { UNSPENDABLE_CONTRACT_ERGO_TREE, gridzDefaults } from "./common";
+import settingsContract from "./contracts/Settings.es";
+import { hex } from "@fleet-sdk/crypto";
+import { mockUTxO } from "@fleet-sdk/mock-chain";
+
+type RequiredFields = {
+  value: bigint;
+  // TODO: probably shouldnt be required if we need a input box?
+  creationHeight: number;
+  ergoTree: ErgoTree;
+};
+
+// similiar concept to "PageObject": https://martinfowler.com/bliki/PageObject.html
+// goal is to abstract away how the boxs properties (registers, tokens, etc) are assembled
+// from client code so the underlying box layout is flexible to change
+export abstract class BoxObject<F> {
```

Comment:
This PR ended up mostly just being adding this box/tx object concept I had being thinking about.

It is meant to be similiar to `PageObject`s: concept to "PageObject": https://martinfowler.com/bliki/PageObject.html with the goal to:

Abstract box layout from client code (i.e clients shouldn't need to know register layouts, they should just need to set a boxes field like `makerFee`) and allow the boxes layout to be flexible to change without needing to update client code